account_asset_management_xls/report/__init__.py

```diff
@@ -1,2 +1,10 @@
 # -*- coding: utf-8 -*-
-from . import account_asset_report_xls
+try:
+    from odoo.addons.report_xlsx_helper.report.abstract_report_xlsx \
+        import AbstractReportXlsx
+except (ImportError, IOError) as err:
+    import logging
+    _logger = logging.getLogger(__name__)
+    _logger.debug(err)
```

Comment:
Shouldn't you import a module here (``account_asset_report_xls``?) to fallback when importing ``AbstractReportXlsx`` has failed?
Ditto for the 2 other files?

Reply:
We do it in the else statement

Reply:
I am not sure to understand the doc https://docs.python.org/3/tutorial/errors.html#handling-exceptions
> The try … except statement has an optional else clause, which, when present, must follow all except clauses. It is useful for code that must be executed if the try clause does not raise an exception.

=> Doesn't it mean that the "else" statement is only run if no exception is caught previously?

Reply:
@asaunier he only loads the local module if its dependencies are present otherwise it doesn't load the import the files. so other import isn't needed since what would need them isn't loaded when they're not present.